tests/persistAsync.test.tsx

```diff
@@ -25,11 +24,7 @@ describe('persist middleware with async configuration', () => {
         {
           name: 'test-storage',
           getStorage: () => storage,
-          onRehydrateStorage: () => (state) => {
-            postRehydrationCallbackCallCount++
-            expect(state?.count).toBe(42)
```

Comment:
As written in the description, those `expect()` calls never worked. They resulted in an error being thrown [here](https://github.com/pmndrs/zustand/blob/master/src/middleware.ts#L363) which only a second `catch()` would catch (I actually tried).

So I'd even go so far as to say that this way of testing callbacks is an anti pattern that should be avoided.